system/ee/legacy/libraries/Redirect.php

```diff
@@ -30,20 +30,17 @@
 
 // If the URL doesnt start with http or https (and the url doesnt start with ://), change the URL before moving on
 if (strncmp($_GET['URL'], 'http', 4) != 0 && strpos($_GET['URL'], '://') === false) {
-    // get the protocol
-    if(isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on') {
-        $protocol = "https://";
-    } else {
-        $protocol = "http://";
-    }
+    $protocol = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on') ? 'https://' : 'http://';
 
     // if the url starts with /, add the http host
     if (substr($_GET['URL'], 0, 1) === '/') {
-        $_GET['URL'] = $_SERVER['HTTP_HOST'] . $_GET['URL'];
+        $_GET['URL'] = ee()->config->site_url($_GET['URL']);
```

Comment:
The site_url and base_url can be set to empty string in config file. So I'm not sure we can rely on those here (and also  we're processing HTTP_HOST later in this file anyway)

Additionally, this is kinda strange function because it's checking for `base_url` and not `site_url` so I wonder it we should completely deprecate it in favor of `ee()->function->create_uri()`